src/units/unit.hpp

```diff
@@ -1897,6 +1897,14 @@ class unit : public std::enable_shared_from_this<unit>
 	 * @param filter the filter used for checking.
 	 */
 	bool ability_matches_filter(const config & cfg, const std::string& tag_name, const config & filter) const;
+	/**
+	 * Verify what unit checked is not itself.
```

Comment:
Please copy the comment from abilities.cpp, it's clearer and includes the "same location" part.